TTS_xwing/src/Game/Mechanic/Movement/Maneuver.ttslua

```diff
@@ -0,0 +1,242 @@
+
+local igu = Dim.Convert_mm_igu
+-- TODO: add reverse move, and reverse bank
+
+----------------------------------------
+-- Standard X-Wing Maneuver Templates --
+----------------------------------------
+
+local XWingTemplates = {
+    straight = { 40, 80, 120, 160, 200 }, -- speed to millimeters
+    turn = { 35, 63, 90 },                -- Turn Middle Radius
+    bank = { 80, 130, 180 },              -- Bank Middle Radius
+}
+
+-----------------------------
+-- Strategy Design Pattern --
+-----------------------------
+
+-- Base MovementStrategy class
+local ManeuverStrategy = {}
+ManeuverStrategy.__index = ManeuverStrategy
+
+function ManeuverStrategy:new()
+    return setmetatable({}, self)
+end
+
+function ManeuverStrategy:execute(ship)
+    -- To be overridden by subclasses
+end
+
+function ManeuverStrategy:getBaseSize(ship)
+    local shipData = ship.getTable("Data")
+    local size = shipData.Size or "small"
+    if type(size) == "string" then
+        return igu(Dim.mm_baseSize[size])
+    elseif type(size) == "number" then
+        return igu(size) -- Assuming the size is provided in mm
+    else
+        error("Invalid size type")
+    end
+end
+
+--[[
+    If the forward Vector of the object doesn't line up with the object
+    visually, you can attach an angle value here to correct the forward vector
+    by rotating it around the y-axis by the provided angle in degrees.
+    This will correct an object's forward vector to move it properly.
+    [+ is clockwise, - is counter-clockwise]
+ ]]
+function ManeuverStrategy:getForwardVector(ship)
+    local shipData = ship.getTable("Data")
+    local forwardAdjustment = shipData.forwardAdjustment or 0
+    local forward = ship.getTransformForward():copy()
+    forward = forward:rotateOver("y", forwardAdjustment)
+    return forward:normalized()
+end
+
+-- StraightMove class inherits from MovementStrategy
+local Straight = setmetatable({}, { __index = ManeuverStrategy })
+Straight.__index = Straight
+
+function Straight:new(speed)
+    local instance = ManeuverStrategy:new()
+    instance.speed = igu(XWingTemplates.straight[speed])
+    return setmetatable(instance, self)
+end
+
+function Straight:execute(ship)
+    local forward = self:getForwardVector(ship)
+    local baseSize = self:getBaseSize(ship)
+    local posDelta = (self.speed + baseSize) * forward
+    local newPos = ship.getPosition() + posDelta
+    ship.setPositionSmooth(newPos, false, true)
+end
+
+-- KoiogranTurn class inherits from StraightMove
+local KoiogranTurn = setmetatable({}, { __index = Straight })
+KoiogranTurn.__index = KoiogranTurn
+
+function KoiogranTurn:new(speed)
+    local instance = Straight:new(speed)
+    instance.rotation = 180
+    return setmetatable(instance, self)
+end
+
+function KoiogranTurn:execute(ship)
+    Straight.execute(self, ship)
+    local newRotation = ship.getRotation() + Vector(0, self.rotation, 0)
+    ship.setRotationSmooth(newRotation, false, false)
+end
```

Comment:
I've not worked with inheritance in LUA, could this not been part of execute straight (where the non k turn version has a rotate of 0?)

Reply:
I kept debating whether the Straight Move should take another argument for rotation rather than Koiogran doing it's own. So far, I opted to keep Straight move simpler. Another requirement could change it.